sources/org.osbuild.curl

```diff
@@ -123,19 +127,27 @@ def fetch(url, checksum, directory):
             if return_code == 0:
                 break
         else:
-            raise RuntimeError(f"curl: error downloading {url}: error code {return_code}")
-
-        if not verify_file(f"{tmpdir}/{checksum}", checksum):
-            raise RuntimeError(f"checksum mismatch: {checksum} {url}")
-
-        # The checksum has been verified, move the file into place. in case we race
-        # another download of the same file, we simply ignore the error as their
-        # contents are guaranteed to be  the same.
-        try:
-            os.rename(f"{tmpdir}/{checksum}", f"{directory}/{checksum}")
-        except FileExistsError:
-            pass
-
+            error = f"curl: error downloading, last return code {return_code}"
+
+        if not error and not verify_file(f"{tmpdir}/{checksum}", checksum):
+            error = f"checksum mismatch: {checksum}"
+
+        if not error:
+            # The checksum has been verified, move the file into place. In case
+            # we race another download of the same file, we simply ignore the
+            # error as their contents are guaranteed to be  the same.
+            try:
+                os.rename(f"{tmpdir}/{checksum}", f"{directory}/{checksum}")
+            except FileExistsError:
+                error = "redundant downloading of the same file"
```

Comment:
This should be considered success, I think.

Reply:
Yes you are right! I transformed the logic here. In my mind I was thinking that maybe we could make it a warning. What do you think ?